Selectively disable executor unit tests for hardware with SVM virtualization

bulwahn opened the pull request:
As discussed on the mailing list (see Link), the executor unit tests fail on some newer hardware, probably due to SVM virtualization not supporting older VMX virtualization features. So, we disable the tests for that. Before touching the file, we made the style checker happy. And afterwards, we add Piotr as contributor and Elektrobit as copyright holder.

Link: https://groups.google.com/g/syzkaller/c/jmb8Pt8qdQg

dvyukov commented on executor/common_kvm_amd64.h:
```diff
@@ -800,3 +800,26 @@ static volatile long syz_kvm_setup_cpu(volatile long a0, volatile long a1, volat
+#define FEATURE_AMD 0x800000001
+#define FEATURE_AMD_ECX_SVM (1 << 2)
+
+static long cpu_feature_enabled(uint32_t function, uint32_t eax_bits, uint32_t ebx_bits, uint32_t ecx_bits, uint32_t edx_bits)
```
This should go in the test file. Otherwise it will appear in syz-executor binary and in C reproducers.

bimberix replied:
Got it. Thanks.

bimberix commented on executor/common_kvm_amd64.h:
```diff
@@ -800,3 +800,26 @@ static volatile long syz_kvm_setup_cpu(volatile long a0, volatile long a1, volat
+	for (uint32_t i = 0; i < cpuid->nent; i++) {
+		struct kvm_cpuid_entry2* entry = &cpuid->entries[i];
+		if (entry->function == function && (!eax_bits || (entry->eax & eax_bits)) && (!ebx_bits || (entry->ebx & ebx_bits)) && (!ecx_bits || (entry->ecx & ecx_bits)) && (!edx_bits || (entry->edx & edx_bits)))
+			return 0;
```
Yes. That would be more intuitive. I will add return status codes in comet as well.

bimberix commented on executor/common_kvm_amd64.h:
```diff
@@ -800,3 +800,26 @@ static volatile long syz_kvm_setup_cpu(volatile long a0, volatile long a1, volat
+
+static long cpu_feature_enabled(uint32_t function, uint32_t eax_bits, uint32_t ebx_bits, uint32_t ecx_bits, uint32_t edx_bits)
+{
+	int kvmfd = open("/dev/kvm", O_RDWR);
+	char buf[sizeof(struct kvm_cpuid2) + 128 * sizeof(struct kvm_cpuid_entry2)];
+	memset(buf, 0, sizeof(buf));
+	struct kvm_cpuid2* cpuid = (struct kvm_cpuid2*)buf;
+	cpuid->nent = 128;
+	ioctl(kvmfd, KVM_GET_SUPPORTED_CPUID, cpuid);
+	close(kvmfd);
```
Will do.

bulwahn commented:
@dvyukov @tarasmadan Piotr has addressed your comments, we reviewed it internally and I have added the two further commits to this pull request to see the changes due to review in a simple way (sorry, for the delay on my side to get this public). If the content is now good, I am happy to squash the fix-up commits, as you like, rebase it all to current master or change this "local git commit history" to anything else as you like. Just let us know if content is good and which commits you would like to keep as own commits in the repository's history.

a-nogikh commented on executor/test_linux.h:
```diff
@@ -263,3 +278,30 @@ static void dump_cpu_state(int cpufd, char* vm_mem)
+{
+	int kvmfd = open("/dev/kvm", O_RDWR);
+	if (kvmfd == -1) {
+		perror("Error opening /dev/kvm");
```
Nit: we don't use `perror()` anywhere else, let's just use printf for consistency.

a-nogikh reviewed (commented):
The code looks good to me. Please squash the fix commits and we'll merge it all.

bulwahn commented:
We fixed the nit and squashed and rebased. With that, it is hopefully good to be merged now.
